src/main/java/com/redhat/cloud/policies/app/auth/models/Access.java

```diff
@@ -18,9 +18,48 @@
 
 import java.util.List;
 
-public class Access {
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class Access extends RbacRawCommon {
+
+    @JsonIgnore
+    public static final String HOST_GROUPS_OPERATION = "in";
+
+    @JsonIgnore
+    public static final String HOST_GROUPS_KEY = "group.id";
 
     public String permission;
     public List<ResourceDefinition> resourceDefinitions;
 
+    public String[] getPermissionFields() {
+        if (permission == null) {
+            return new String[3];
+        }
+        return permission.split(":");
+    }
+
+    public boolean isInventoryHostsRead() {
+        String[] permissionFields = getPermissionFields();
+        return permissionFields[0].equals(INVENTORY_HOSTS_READ[0])
+            && (permissionFields[1].equals(INVENTORY_HOSTS_READ[1]) || permissionFields[1].equals(ANY))
+            && (permissionFields[2].equals(INVENTORY_HOSTS_READ[2]) || permissionFields[2].equals(ANY));
```

Comment:
This will throw a NullPointerException if `permission` is null as each element of the `permissionFields` will be null.

Use [Objects.equals](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-) instead

You will need to include `java.util.Objects`

```suggestion
        return Objects.equals(permissionFields[0], INVENTORY_HOSTS_READ[0])
            && (Objects.equals(permissionFields[1], INVENTORY_HOSTS_READ[1]) || Objects.equals(permissionFields[1], ANY))
            && (Objects.equals(permissionFields[2], INVENTORY_HOSTS_READ[2]) || Objects.equals(permissionFields[2], ANY));
```

Reply:
Updated. Thanks.